tests/run_test.sh

```diff
@@ -299,8 +299,16 @@ if (( NODES * TPN < TASKS )); then
 fi
 export NODES
 
-TASKS=$(( NODES * TPN ))
-export TASKS
+PPN=$(( (TASKS - 1)/NODES + 1 ))
+
+if (( $(( PPN % 2 )) > 0 )); then
+  PPN=$(( PPN + 1 ))
+fi
+
+if (( PPN > TPN )); then
+  PPN=${TPN}
+fi
+export PPN
```

Comment:
What is this block of code doing? Why it is necessary? Is this something something specific to Derecho? What's the meaning of PPN, 'process per node'?  Isn't TPN already equal to 'process per node'?

Reply:
@mark-a-potts @zach1221 can you answer Dusan's question?

Reply:
This is the explanation(s) I have. I think this is Derecho specific, but I can test again with the original setup to see if there are issues.

**// determine # of processes per node to be used with mpiexec/mpirun/srun**
PPN=$(( (TASKS - 1)/NODES + 1 ))
**// If PPN is odd, add one to make even.**
if (( $(( PPN % 2 )) > 0 )); then
  PPN=$(( PPN + 1 ))
fi

**// If more processes per node than cores, set PPN = TPN.**
if (( PPN > TPN )); then
  PPN=${TPN}
fi
export PPN
**//  PPN may then be used to distribute mpi tasks in the mpirun/mpiexec/srun command, but this is uncertain**

Reply:
Ok, I added the original line below back to run_test.sh , as well as added -ppn @[TPN] to line 38 of fv3_conf/fv3_qsub.IN_derecho

Added back to run_test.sh:
TASKS=$(( NODES * TPN ))
export TASKS

working directory: /glade/work/zshrader/rt-1836/tests/logs/RegressionTests_derecho.log. There were two failures but those were wallclock timeouts. So, I can commit those code changes back to the PR, since they don't appear to cause issues.

Reply:
Both of the time-outs were actually hangs I think. Both only get to the 3 min Advance time, which is the dt_atmos=180s.

Reply:
@DeniseWorthen yes you're right. Hanging with those two cases appears to be persistent if we're to revert run_test.sh back to the original.

Reply:
The two tests which time-out with the original run_test.sh were ``hafs_regional_atm_ocn_wav_intel`` and ``hafs_regional_storm_following_1nest_atm_ocn_wav_intel``. Is that right?

What is the difference in the job-cards (if any) for those tests w/ the two versions of run_test.sh?

Reply:
No. TPN should have either initial value specified in the default-vars.sh (specific to each platform), or the value overwritten by individual test/platform combination (to accommodate cases which require more memory). After this 'initial' value is defined we set it to TPN = TPN / THRD, where THRD is number of OpenMP threads.

Then, number of nodes is computed as NODES = TASKS / TPN.

So, total number of tasks (TASKS) and tasks per node (TPN) determine number of nodes (NODES).